test/unit/rules/lint-no-element-event-actions-test.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+const ERROR_MESSAGE = require('../../../lib/rules/lint-no-element-event-actions').ERROR_MESSAGE;
+
+generateRuleTests({
+  name: 'no-element-event-actions',
+
+  config: true,
+
+  good: [
+    '<button></button>',
+    '<button type="button" on={{action "myAction"}}></button>',
+    '<button type="button" onclick="myFunction()"></button>',
+    '<button type="button" {{action "myAction"}}></button>',
+    '<button type="button" value={{value}}></button>',
+    '{{my-component onclick=(action "myAction") someProperty=true}}',
+    '<SiteHeader click={{action "myAction"}} @user={{this.user}} />',
```

Comment:
Any of the things where the attribute name is not prefixed by `@` will be set as attribute / property (depending on the name and whatnot like I mentioned before) on whatever element has `...attributes` on it. 

If the idea here is to pass in a click function that the component as an argument that it can use in its JS file (ala `this.click()`) then you would do:

```suggestion
    '<SiteHeader @click={{action "myAction"}} @user={{this.user}} />',
```

Reply:
Updated!